en/news/_posts/2018-11-08-snap.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: news_post
+title: The official ruby snap is available
+author: Hiroshi SHIBATA
+translator:
+date: 2018-11-08 14:58:28 +0000
+lang: en
+---
+
+We released the official snap package of Ruby language.
+
+https://snapcraft.io/ruby
+
+snap is the package system developed by Canonical. You can distribute the package with necessary libraries by using snap. It solved the problem that users can not use any versions of Ruby interpreter　when using Linux distribution package system like rpm or apt.
+
+For Ubuntu 16.04 or later distribution, you can use Ruby snap with the following command.
+
+    sudo snap install ruby -classic
+
+When you use the Linux distributions other than Ubuntu, Please refer to https://docs.snapcraft.io/installing-snapd/6735.
+
+Ruby snap uses the feature called channel to distribute the currently maintained version. For example, if you do not specify a channel at 2018/11, It will be installed 2.5.3. but if you want to use Ruby 2.4, specify it as follows.
+
+    sudo snap install ruby --classic --channel=2.4/stable
+
+Multiple Ruby versions can be used with the snap feature. For example, to switch to Ruby 2.3, execute the following command.
+
+    sudo snap switch ruby --channel=2.3/stable
+    sudo snap refresh
+
+The Ruby snap has set `$HOME/.gem` to `GEM_HOME` and `GEM_PATH` environmental variables. Therefore, if you want to execute commands such as rails and rspec without using `bundle exec`, you need to set the following lines to` .bashrc` and so on.
+
+    eval `ruby.env`
+
+Since `$HOME/.gem` is shared by multiple versions, if you switch versions and use them, you will need to recompile C extensions using the `gem pristine` command.
```

Comment:
Is this true? I think extension library files should live under `~/.gem/extensions/x86_64-linux/$ABI`.

Reply:
>C extension だけ見つからないという挙動になるので pristine が必要
>(Since the extension files (.so, .bundle) are only unshared between different Ruby ABI version (unlike .rb files,) this is true)

Understood, yes, I missed the normal .rb files are still shared between rubies. So:

> Since `$HOME/.gem` is shared by multiple Ruby versions, you may need to run `gem pristine` to build extension libraries when you switched to a different channel for the first time.